presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestEliminateSorts.java

```diff
@@ -76,12 +76,10 @@ public void testNotEliminateSorts()
         PlanMatchPattern pattern =
                 anyTree(
                         sort(
-                                anyTree(
-                                        window(windowMatcherBuilder -> windowMatcherBuilder
-                                                        .specification(windowSpec)
-                                                        .addFunction(functionCall("row_number", Optional.empty(), ImmutableList.of())),
-                                                anyTree(LINEITEM_TABLESCAN_Q)))));
-
+                                window(windowMatcherBuilder -> windowMatcherBuilder
```

Comment:
curious: why remove `anyTree`?

Reply:
anyTree doesn't match zero nodes.  there must have been an intervening node that got removed after the refactor.  I don't recall what it was, but I would guess it's a remote exchange due to the slightly different logic and this test running on a single node.

Reply:
It feels weird when a refactor needs to touch tests. As you mentioned, the logic is different because there are cases that child is single node but does not hit `return` (https://github.com/prestodb/presto/pull/11957/commits/73c23b0d4f325c56093e882a8d972177c3d205d2#diff-3147362e4362824887f1eb890a167d41L428), which might end up having distributed sort enabled. This is not covered after refactoring.

I'm not sure if this would make the plan less optimal or more optimal, but if we could eliminate a remote exchange, maybe the purpose of this commit change from "refactor" to "optimize"?